benchmark/benchresult/main.go

```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+To format the benchmark result:
+  go run benchmark/benchresult/main.go resultfile
+
+To see the performance change based on a old result:
+  go run benchmark/benchresult/main.go resultfile_old resultfile
+It will print the comparison result of intersection benchmarks between two files.
+
+*/
+package main
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/benchmark/stats"
+)
+
+func createMap(fileName string, m map[string]stats.BenchResults) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Read file %s error: %s\n", fileName, err)
+	}
+	defer f.Close()
+	var data []stats.BenchResults
+	decoder := gob.NewDecoder(f)
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	for _, d := range data {
+		m[d.RunMode+"-"+d.Features.String()] = d
+	}
+}
+
+func intChange(title string, val1, val2 int64) string {
+	percentChange := strconv.FormatFloat(float64((val2-val1)*100/val1), 'f', 2, 64) + "% "
```

Comment:
Why a trailing space?

And why the trailing space in the sprintf below?

You shouldn't need this anyway.. You should be able to do:
```go
fmt.Sprintf(".... %8.2f", ..., (100*(val2-val1))/val1)
```
Same for the FormatInt.  Use %v or %d to format it via the Sprintf.

Reply:
Done